modules/elasticsearch-index/src/main/java/org/opencastproject/elasticsearch/index/ElasticsearchIndex.java

```diff
@@ -130,15 +139,39 @@ public void activate(ComponentContext ctx) throws ComponentException {
   /**
    * OSGi callback to deactivate this component.
    *
-   * @param ctx
-   *          the component context
    * @throws IOException
+   *          If closing the index fails
    */
   @Deactivate
-  public void deactivate(ComponentContext ctx) throws IOException {
+  public void deactivate() throws IOException {
     close();
   }
 
+  /**
+   * OSGi callback for configuration changes.
+   *
+   * @param properties
+   *          The configuration
+   */
+  @Modified
+  public void modified(Map<String, Object> properties) {
+    super.modified(properties);
+
+    maxRetryAttemptsGet = NumberUtils.toInt((String) properties.get(MAX_RETRY_ATTEMPTS_GET_PROPERTY),
```

Comment:
Out of curiosity, why are these here rather than in the superclass?

Reply:
The reasoning was that properties that are used by the super class will get read in by that one, while everything else will be left to the sub class.